packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -325,3 +315,94 @@ function getUpdatedWorkspaceWindowState(workspaceAtTop: OpenWorkspace) {
 }
+
+type PromptType = 'closing-workspace' | 'closing-all-workspaces' | 'closing-workspace-launching-new-workspace';
+
+export function showWorkspacePrompts(
+  promptType: PromptType,
+  onConfirmation: () => void = () => {},
+  workspaceTitle: string = '',
+) {
+  const store = getWorkspaceStore();
+
+  switch (promptType) {
+    case 'closing-workspace': {
+      const prompt: Prompt = {
+        title: translateFrom('@openmrs/esm-patient-chart-app', 'unsavedChangesTitleText', 'Unsaved Changes'),
+        body: translateFrom(
+          '@openmrs/esm-patient-chart-app',
+          'unsavedChangeText',
+          `You have unsaved changes in the side panel. Do you want to discard these changes?`,
+        ),
+        onConfirm: () => {
+          onConfirmation?.();
+        },
+        confirmText: translateFrom('@openmrs/esm-patient-chart-app', 'discard', 'Discard'),
+      };
+      store.setState({ ...store.getState(), prompt });
+      store.setState((prevState) => ({ ...prevState, prompt }));
+      return;
+    }
+
+    case 'closing-all-workspaces': {
+      const workspacesNotClosed = store
+        .getState()
+        .openWorkspaces.filter(({ name }) => !getWhetherWorkspaceCanBeClosed(name))
+        .map(({ title }, indx) => `${indx + 1}. ${title}`);
+
+      const prompt: Prompt = {
+        title: translateFrom('@openmrs/esm-patient-chart-app', 'unsavedChanges', 'You have unsaved changes'),
+        body: translateFrom(
+          '@openmrs/esm-patient-chart-app',
+          'unsavedChangesInForms',
+          `There are unsaved changes in the following workspaces. Do you want to discard changes in the following workspaces? {{workspaceNames}}`,
+          {
+            workspaceNames: workspacesNotClosed.join(' '),
+          },
+        ),
+        onConfirm: () => {
+          onConfirmation?.();
+        },
+        confirmText: translateFrom(
+          '@openmrs/esm-patient-chart-app',
+          'closeWorkspaces',
+          'Discard changes in {{count}} workspaces',
+          { count: workspacesNotClosed.length },
+        ),
+      };
+      store.setState((prevState) => ({
+        ...prevState,
+        prompt,
+      }));
+      return;
+    }
+    case 'closing-workspace-launching-new-workspace': {
+      const prompt: Prompt = {
+        title: translateFrom('@openmrs/esm-patient-chart-app', 'unsavedChanges', 'You have unsaved changes'),
+        body: translateFrom(
+          '@openmrs/esm-patient-chart-app',
+          'unsavedChangesInForm',
+          'There are unsaved changes in {{formName}}. Please save them before opening another form.',
+          { formName: workspaceTitle },
+        ),
+        onConfirm: () => {
+          store.setState((prevState) => ({
+            ...prevState,
+            prompt: null,
+          }));
+          onConfirmation?.();
+        },
+        confirmText: translateFrom('@openmrs/esm-patient-chart-app', 'openAnyway', 'Open anyway'),
```

Comment:
This text should mention that the previous form will be discarded, e.g., "Discard changes and open anyway"